### ISSUE TITLE
Flexible heredoc nog recognized

### ISSUE THREAD

**Issue by tacovandenbroek** [kalessil/phpinspectionsea#1543]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) version 4.0.3 |
| Language level | PHP 7.3                        |

Current behaviour
--
![2020-04-09_10-48](https://user-images.githubusercontent.com/1499481/78877242-d5bb0f80-7a50-11ea-9462-eb80d3488a9f.png)

Expected behaviour
--
I'd expect no warning for this valid regular expression.

Environment details
---
PhpStorm 2019.3

**Comment by kalessil** [comment]
Thank you for reporting @tacovandenbroek, I'll look into it.

**Comment by kalessil** [comment]
@tacovandenbroek : unfortunately I was not able to reproduce the issue. Would you please share an isolated code fragment where the issue appears?

**Comment by tacovandenbroek** [comment]
Sure! It seems to depend on the context. I couldn't trigger the inspection without wrapping the code in a function:

```php
<?php
function doesMatch(string $subject): bool {
    $regex = <<< REGEX
            (foo)
            REGEX;
    return (bool)preg_match($regex, $subject);
}
```

**Comment by ausi** [comment]
I think this issue is caused by leading white space (even though the heredoc example does not have the whitespace in the resulting string). 

I was able to reproduce it with this simple test case:

```php
preg_match(' /.+/', '');
```
<img width="692" alt="Bildschirmfoto 2022-02-24 um 12 44 44" src="https://user-images.githubusercontent.com/367169/155518278-10be1222-bbb5-4710-9965-dfd072212fff.png"><br>PhpStorm 2021.3.2, PHP Inspections (EA Extended) 4.0.7.1

Leading an trailing whitespace is supported in PCRE, see <https://3v4l.org/Kvcqa>

**Comment by ausi** [comment]
Maybe changing the lines from here https://github.com/kalessil/phpinspectionsea/blob/9bd52620388ca5f8f0d894d49ad5e921a6adf533/src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/regularExpressions/NotOptimalRegularExpressionsInspector.java#L67-L71 to something like this (adding `\s*` to the beginning and end) could work?

```java
matchers.add(Pattern.compile("^\\s*([^{<(\\[])(.*)(\\1)([a-zA-Z]+)?\\s*$", Pattern.DOTALL));
matchers.add(Pattern.compile("^\\s*(\\{)(.*)(\\})([a-zA-Z]+)?\\s*$", Pattern.DOTALL));
matchers.add(Pattern.compile("^\\s*(<)(.*)(>)([a-zA-Z]+)?\\s*$", Pattern.DOTALL));
matchers.add(Pattern.compile("^\\s*(\\()(.*)(\\))([a-zA-Z]+)?\\s*$", Pattern.DOTALL));
matchers.add(Pattern.compile("^\\s*(\\[)(.*)(\\])([a-zA-Z]+)?\\s*$", Pattern.DOTALL));
```
